### PR TITLE
amtk: remove clang dependency

### PR DESCRIPTION
**Pull request by satmandu**
- compiles just fine with gcc, with meson lld cflag.

Works properly:
- [x] x86_64

### CODE REVIEW

**Comment by cstrouse** [comment]
@satmandu CrOS is built with llvm. What is the rationale for wanting to change packages to use gcc?

**Comment by uberhacker** [comment]
@cstrouse: I think he wants to avoid the use of `ENV` variables.  @satmandu can elaborate more on this.

**Comment by satmandu** [comment]
@cstrouse I have no complaints about llvm (aside from how painful I am finding builting llvm 11  with lto=thin). I still have these packages using llvm's lld linker. But I'm seeing ENV issues where ```['CC'] = 'clang'``` is set and then subsequent package installs have problems because they think the CC environment variable is STILL set to clang, even if I literally have a ```['CC'] = 'gcc'``` section in the second package file.

So I'm just going through and getting rid of all of the packages which have CC set to clang or CXX set to clang++.

If a compile with GCC fails, we can also set ```CC=clang ./configure...``` or whatever.

(But I am also wondering if ENV should be used for setting environment variables at all. Isn't that frowned upon in ruby?)


TL;DR: Not actually avoiding using all llvm components, as we are still often using lld. Just avoiding setting packages to use clang or clang++ through ENV settings where possible to rule out that being an issue with package compile failures.

**Comment by cstrouse** [comment]
That makes sense. I just wanted to make sure we capture the rationale for things in the commit messages or code reviews for reference by future contributors. Thanks for the clarification @satmandu

**Comment by satmandu** [comment]
Here's a place which mentions that one should only use ENV to get environment variables:
https://www.rubyguides.com/2019/01/ruby-environment-variables/#menu-item-3157:~:text=Warning%3A,like%20dry%2Dconfigurable%20for%20internal%20configuration%20options.

image
